homeassistant/components/ccm15/coordinator.py

```diff
@@ -0,0 +1,247 @@
+        """Fetch data from Rain Bird device."""
+        try:
+            return await self._fetch_data()
+        except httpx.RequestError as err:  # pragma: no cover
+            raise UpdateFailed(f"Error communicating with Device: {err}") from err
+
+    async def _fetch_data(self) -> CCM15DeviceState:
+        """Get the current status of all AC devices."""
+        ac_data = await self._ccm15.get_status_async()
+        if len(self._ac_devices) == 0:
+            for ac_index in ac_data.devices:
+                _LOGGER.debug("Creating new ac device at index '%s'", ac_index)
+                self._ac_devices[ac_index] = CCM15Climate(self._host, ac_index, self)
+        return ac_data
+
+    async def async_test_connection(self):  # pragma: no cover
+        """Test the connection to the CCM15 device."""
+        return await self._ccm15.async_test_connection()
+
+    async def async_set_state(self, ac_index: int, state: str, value: int) -> None:
+        """Set new target states."""
+        if await self._ccm15.async_set_state(ac_index, state, value):
+            await self.async_request_refresh()
+
+    def get_ac_data(self, ac_index: int) -> CCM15SlaveDevice | None:
+        """Get ac data from the ac_index."""
+        if ac_index < 0 or ac_index >= len(self.data.devices):
+            # Index is out of bounds or not an integer
+            _LOGGER.warning(
+                "Data is not available for AC id %s.%s", self._host, ac_index
+            )
+            return None
+        data = self.data.devices[ac_index]
+        return data
+
+    async def async_set_hvac_mode(self, ac_index, hvac_mode: HVACMode) -> None:
+        """Set the hvac mode."""
+        _LOGGER.debug("Set Hvac[%s]='%s'", ac_index, str(hvac_mode))
+        await self.async_set_state(ac_index, "mode", CONST_STATE_CMD_MAP[hvac_mode])
+
+    async def async_set_fan_mode(self, ac_index, fan_mode: str) -> None:
+        """Set the fan mode."""
+        _LOGGER.debug("Set Fan[%s]='%s'", ac_index, fan_mode)
+        await self.async_set_state(ac_index, "fan", CONST_FAN_CMD_MAP[fan_mode])
+
+    async def async_set_temperature(self, ac_index, temp) -> None:
+        """Set the target temperature mode."""
+        _LOGGER.debug("Set Temp[%s]='%s'", ac_index, temp)
+        await self.async_set_state(ac_index, "temp", temp)
+
+
+class CCM15Climate(CoordinatorEntity[CCM15Coordinator], ClimateEntity):
+    """Climate device for CCM15 coordinator."""
+
+    def __init__(
+        self, ac_host: str, ac_index: int, coordinator: CCM15Coordinator
+    ) -> None:
+        """Create a climate device managed from a coordinator."""
+        super().__init__(coordinator)
+        self._ac_host: str = ac_host
+        self._ac_index: int = ac_index
+        self._attr_should_poll = True
+        self._attr_temperature_unit = UnitOfTemperature.CELSIUS
+        self._attr_has_entity_name = True
+        self._attr_name = f"{self._ac_index}"
```

Comment:
The name of the entity is a number? That's not user-friendly. Do you think we can use something else?

Reply:
The ccm15 module treats each AC unit as a zero-based index. So, there are really not good options in terms of naming that could provide anything useful to the end user other than the index.